Integration test is flaky for status updates

see https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/90/pull-kueue-test-integration-main/1501214768197799936

Managed to reproduce locally after several runs 
```bash
    STEP: Creating workloads 03/09/22 16:48:04.446
  << End Captured GinkgoWriter Output

  Timed out after 10.001s.
  Mismatch (-want,+got):
    v1alpha1.QueueStatus{
  -     PendingWorkloads: 3,
  +     PendingWorkloads: 2,
    }
  
  In [It] at: /home/eduardo/sdk/github/kubernetes-sigs/kueue/test/integration/controller/core/queue_controller_test.go:84
------------------------------
•

Ran 2 of 2 Specs in 17.084 seconds
FAIL! -- 1 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (17.08s)
```

Seems to alight with Aldo idea of it being around the Queue controller

--- a/pkg/constants/constants.go
+++ b/pkg/constants/constants.go
@@ -16,6 +16,8 @@ limitations under the License.
 
 package constants
 
+import "time"
+
 const (
 	// QueueAnnotation is the annotation in the workload that holds the queue name.
 	//
@@ -24,4 +26,8 @@ const (
 
 	ManagerName       = "kueue-manager"
 	JobControllerName = "kueue-job-controller"
+
+	// UpdatesBatchPeriod is the batch period to hold queuedworkload updates
+	// before syncing a Queue and CLusterQueue onbjects.
+	UpdatesBatchPeriod = 3 * time.Second
 )

--- a/pkg/controller/core/clusterqueue_controller.go
+++ b/pkg/controller/core/clusterqueue_controller.go
@@ -30,6 +30,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+	"sigs.k8s.io/kueue/pkg/constants"
 	"sigs.k8s.io/kueue/pkg/queue"
 
 	kueue "sigs.k8s.io/kueue/api/v1alpha1"
@@ -145,25 +146,25 @@ type assignedWorkloadHandler struct{}
 func (h *assignedWorkloadHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
 	w := e.Object.(*kueue.QueuedWorkload)
 	if w.Spec.Admission != nil {
-		q.Add(requestForWorkloadClusterQueue(w))
+		q.AddAfter(requestForWorkloadClusterQueue(w), constants.UpdatesBatchPeriod)
 	}
 }
 
 func (h *assignedWorkloadHandler) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
 	oldW := e.ObjectOld.(*kueue.QueuedWorkload)
 	if oldW.Spec.Admission != nil {
-		q.Add(requestForWorkloadClusterQueue(oldW))
+		q.AddAfter(requestForWorkloadClusterQueue(oldW), constants.UpdatesBatchPeriod)
 	}
 	newW := e.ObjectNew.(*kueue.QueuedWorkload)
 	if newW.Spec.Admission != nil && (oldW.Spec.Admission == nil || newW.Spec.Admission.ClusterQueue != oldW.Spec.Admission.ClusterQueue) {
-		q.Add(requestForWorkloadClusterQueue(newW))
+		q.AddAfter(requestForWorkloadClusterQueue(newW), constants.UpdatesBatchPeriod)
 	}
 }
 
 func (h *assignedWorkloadHandler) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
 	w := e.Object.(*kueue.QueuedWorkload)
 	if w.Spec.Admission != nil {
-		q.Add(requestForWorkloadClusterQueue(w))
+		q.AddAfter(requestForWorkloadClusterQueue(w), constants.UpdatesBatchPeriod)
 	}
 }
 

--- a/pkg/controller/core/queue_controller.go
+++ b/pkg/controller/core/queue_controller.go
@@ -31,6 +31,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 
 	kueue "sigs.k8s.io/kueue/api/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/constants"
 	"sigs.k8s.io/kueue/pkg/queue"
 )
 
@@ -131,23 +132,23 @@ type queuedWorkloadHandler struct{}
 func (h *queuedWorkloadHandler) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
 	qw := e.Object.(*kueue.QueuedWorkload)
 	if qw.Spec.Admission == nil {
-		q.Add(requestForQueueStatus(qw))
+		q.AddAfter(requestForQueueStatus(qw), constants.UpdatesBatchPeriod)
 	}
 }
 
 func (h *queuedWorkloadHandler) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
 	oldQW := e.ObjectOld.(*kueue.QueuedWorkload)
 	newQW := e.ObjectOld.(*kueue.QueuedWorkload)
-	q.Add(requestForQueueStatus(newQW))
+	q.AddAfter(requestForQueueStatus(newQW), constants.UpdatesBatchPeriod)
 	if newQW.Spec.QueueName != oldQW.Spec.QueueName {
-		q.Add(requestForQueueStatus(oldQW))
+		q.AddAfter(requestForQueueStatus(oldQW), constants.UpdatesBatchPeriod)
 	}
 }
 
 func (h *queuedWorkloadHandler) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
 	qw := e.Object.(*kueue.QueuedWorkload)
 	if qw.Spec.Admission == nil {
-		q.Add(requestForQueueStatus(qw))
+		q.AddAfter(requestForQueueStatus(qw), constants.UpdatesBatchPeriod)
 	}
 }
 

